### ISSUE TITLE
vcfgo stricter than VCF spec in terms of sample filters

### ISSUE THREAD

**Issue by carbocation** [brentp/vcfgo#16]
`vcfgo` seems to expect the sample filter to be the same for all samples:

https://github.com/brentp/vcfgo/blob/aeb512d1a0e94ff2c186a427f50ae01472ac2905/header.go#L69-L73

However, the VCF spec is more permissive: "Trailing fields can be dropped (with the exception of the GT field, which should always be present if specified in the FORMAT field)." See [page 6](https://samtools.github.io/hts-specs/VCFv4.2.pdf).

I believe that, with the exception of GT, trailing fields should be allowed to be missing.

**Comment by brentp** [comment]
Hmm. I haven't seen any VCFs like this in the wild and I'm not sure I want to promote that as it is kinda messy. I could be convinced, especially by a pull request.

**Comment by carbocation** [comment]
Agreed, it's messy and doesn't save enough data to feel rational. But, I have an in-the-wild VCF produced by GATK at the Broad Institute. Am working on a pull request!
